### PR TITLE
Implemented Android HTTP Plugin

### PR DESCRIPTION
**Pull request by v-hoccha**
Fixed and improved Andoird HTTP Plug-in.
Fixed and improved Java Bridge code.

### CODE REVIEW

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java [hunk 1]
```diff
@@ -30,47 +31,79 @@
...
 
     public HttpRequest() {
-        requestBuilder = new Request.Builder();
-        requestCompleted = false;
+        this.requestBuilder = new Request.Builder();
     }
 
-    public static HttpRequest createRequest() {
-        return new HttpRequest();
+    public void setMethod(String method) {
```
Comment (alex-xbox) [inline build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: We only using POST, why we need it?

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java [hunk 2]
```diff
@@ -30,47 +31,79 @@
...
+            return false;
+        }
+        return true;
     }
 
     public void setHeader(String name, String value) {
-        this.requestBuilder = requestBuilder.addHeader(name, value);
+        try
+        {
+            if(name.equalsIgnoreCase("Content-Type"))
```
Comment (alex-xbox) [inline build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: are we planning on using some other content-type except JSON?

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java [hunk 3]
```diff
@@ -30,47 +31,79 @@
...
-            }
-
-            @Override
-            public void onResponse(Call call, final Response response) throws IOException {
-                //if (!response.isSuccessful()) {
-                //    throw new IOException("Unexpected code " + response);
-                //} else {
+    public boolean sendRequest() {
+        try
+        {
```
Comment (alex-xbox) [inline build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: We already have exceptions inside of each call. Do we need this try { catch { block?

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java (hunk 1 above)
Comment (v-hoccha) [reply build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: This is bridge file and it is usually transparently working. And the method can be change anytime if the PlayFab api is changed.So this should be decided in PlayFab side not bridge.

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java (hunk 2 above)
Comment (v-hoccha) [reply build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: This is bridge file and it is usually transparently working. This should be decide in PlayFab side.

File: build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java (hunk 3 above)
Comment (v-hoccha) [reply build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java]: This calling throws exception when the fail create request.

**Pull request merged**